Aspire Budgeting/Facilities/GoogleSheetsManager.swift

```diff
@@ -2,51 +2,70 @@
 //  GoogleSheetsManager.swift
 //  Aspire Budgeting
 //
-//  Created by TeraMo Labs on 11/1/19.
-//  Copyright © 2019 TeraMo Labs. All rights reserved.
-//
 
+import Combine
 import Foundation
 import GoogleAPIClientForREST
 import GTMSessionFetcher
 import os.log
 
-extension Notification.Name {
-  static let hasSheetInDefaults = Notification.Name("hasSheetInDefaults")
+protocol RemoteFileReader {
+  func read(file: File,
+            user: User,
+            location: String) -> AnyPublisher<AnyObject, Error>
+}
+
+protocol RemoteFileWriter {
+  func write(file: File, user: User, location: String)
 }
 
-protocol AspireUserDefaults {
-  func data(forKey defaultName: String) -> Data?
-  func set(_ value: Any?, forKey defaultName: String)
-  func removeObject(forKey defaultName: String)
+typealias RemoteFileReaderWriter = RemoteFileReader & RemoteFileWriter
+
+enum Result<T> {
+  case success(T)
+  case failure(Error)
 }
 
-extension UserDefaults: AspireUserDefaults {}
+final class GoogleSheetsManager: ObservableObject, RemoteFileReaderWriter {
+  func read(file: File,
+            user: User,
+            location: String) -> AnyPublisher<AnyObject, Error> {
+
+    let future = Future<AnyObject, Error> { promise in
+      self.fetchData(spreadsheet: file,
+                     spreadsheetRange: location,
+                     authorizer: user.authorizer as? GTMFetcherAuthorizationProtocol) { (valueRange, error) in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/d5353e560168880237460fc75a2a6ee44e50449b/Aspire Budgeting/Facilities/GoogleSheetsManager.swift#L37"></span>Line should be 100 characters or less: currently 110 characters<br />
<code>line_length</code> <code>GoogleSheetsManager.swift:37</code></td>
    </tr>
  </tbody>
</table>